### PR TITLE
chore: migrate to ESLint flat config

### LINKED ISSUE #25351: Upgrade ESLint config to v9 config system
### Describe the feature

From the ESLint docs:
> We are transitioning to a new config system in ESLint v9.0.0. The config system shared on this page is currently the default but will be deprecated in v9.0.0.

There is no timetable for v9 release ([see this blog post](https://eslint.org/blog/2023/11/whats-coming-in-eslint-9.0.0/#when-to-expect-eslint-v9.0.0)), however the new config system is "feature complete but not enabled by default" therefore, it seems desireable to think about switching to the new config system. 

See: 
- https://eslint.org/docs/latest/use/configure/configuration-files
- https://eslint.org/docs/latest/use/configure/configuration-files-new
- https://eslint.org/blog/2023/11/whats-coming-in-eslint-9.0.0
- https://github.com/eslint/eslint/discussions/17468

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by luc122c**
### 🔗 Linked issue
Resolves #25351

### 📚 Description

As discussed in #25351, eslint are upgrading their config system in eslint v9. The new syntax is available now but is opt-in. We can opt in by replacing `.eslintrc` with a `eslint.config.js` file or setting the environment variable `ESLINT_USE_FLAT_CONFIG=true` before running eslint.

I have followed eslint's [configuration migration guide](https://eslint.org/docs/latest/use/configure/migration-guide) to replace the configuration. This PR is intended to be a straight swap, not modifying the configuration in any way, just upgrading the syntax. There may be additional work needed to take advantage of the benefits the new syntax brings. 

### 🧪 How to test

1. Pull the branch
2. Install dependencies
3. Run `ESLINT_USE_FLAT_CONFIG=true pnpm run lint`

### ⬇️ Dependencies 

Nuxt depends on several configs and plugins. Below is the status of flat config support for each dependency. ([Data sourced from tracker](https://github.com/eslint/eslint/issues/18093))

| Name | Support | Issue/PR |
| ------ | -------- | --------- |
| [jsdoc](https://github.com//gajus/eslint-plugin-jsdoc) | ✅ | https://github.com/gajus/eslint-plugin-jsdoc/pull/1177 |
| import | ? |  https://github.com/import-js/eslint-plugin-import/pull/2714 |
[unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) | ✅ | https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1886 |
| [no-only-tests](https://github.com/levibuzolic/eslint-plugin-no-only-tests) | ? | 
| [@nuxt/eslint-config](https://github.com/nuxt/eslint/) | 🚧 | https://github.com/nuxt/eslint/releases/tag/v0.3.0-beta.7 | 
| [eslint-config-standard](https://github.com/standard/eslint-config-standard) | 📦 🛑 | https://github.com/standard/eslint-config-standard/issues/351 | 

### ⚠️ Issues

Current issues:
- Standard needs to release a new package
- @nuxt/eslint-config is still in beta

### CODE REVIEW

**Comment by luc122c** [comment]
@antfu Do you have any guidance for getting the `@nuxt/eslint-config` working 🙏

**Comment by antfu** [comment]
For `@nuxt/eslint-config` you will need the `v0.3.0-beta.7` version

**Review by antfu — APPROVED** [review]
LGTM, I think this could be a good starting point. I will refactor them out and move some rules into `@nuxt/eslint-config` in future PRs.

File: package.json [hunk 1]
```diff
@@ -44,6 +44,8 @@
     "magic-string": "^0.30.8"
   },
   "devDependencies": {
+    "@eslint/eslintrc": "^3.0.2",
+    "@eslint/js": "^8.57.0",
```
Comment (DamianGlowala) [inline package.json:48]:
```suggestion
    "@eslint/eslintrc": "3.0.2",
    "@eslint/js": "8.57.0",
```

**Pull request merged**